### PR TITLE
feat(windows): add custom config_win.example.lua

### LINKED ISSUE #1672: [Bug]: Telescope doesn't work on Windows
### Problem description

Both the **Find File** and **Find Word** items on the Dashboard don't seem to work as intended on Windows.

- **Find File**: Doesn't do anything when typing.
- **Find Word**: Generates an error shown below.

Based on `:checkhealth`, it looks like Telescope is missing key dependencies:
- `BurntSushi/ripgrep`
- `sharkdp/fd`

I'm not sure why these didn't install.

---

Also, would it be useful to add a **Windows** label for bugs specifically related to usage on Windows OS for improved searchability? When looking through bugs in other repos related to NeoVim plugins, it has often been hard to distinguish bugs on Windows platforms and bugs relating to the NeoVim window.

### LunarVim version

Commit: 1078f43 (Rolling)

### Neovim version (>= 0.5)

0.5.0

### Relevant log output

```shell
health#which_key#check
========================================================================
  - ERROR: Failed to run healthcheck for "which_key" plugin. Exception:
    function health#check[21]..health#which_key#check, line 1
    Vim(lua):E5108: Error executing lua ...te\pack\packer\opt\which-key.nvim\lua\which-key\keys.lua:525: Invalid buffer id: 24

health#telescope#check
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - ERROR: rg: not found. `live-grep` finder will not function without [BurntSushi/ripgrep](https://github.com/BurntSushi/ripgrep) installed.
  - WARNING: fd: not found. Install [sharkdp/fd](https://github.com/sharkdp/fd) for extended capabilities

## ===== Installed extensions =====

## Telescope Extension: `projects`
  - INFO: No healthcheck provided

health#nvim_treesitter#check
========================================================================
## Installation
  - WARNING: `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
  - OK: `node` found v16.7.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - ERROR: `cc` executable not found.
    - ADVICE:
      - Check that any of { "cl" } is in your $PATH or set the environment variable CC or `require"nvim-treesitter.install".compilers` explicitly!
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - css            ✓ . ✓ ✓ ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - java           ✓ ✓ . ✓ ✓ 
  - javascript     ✓ ✓ ✓ ✓ ✓ 
  - json           ✓ ✓ ✓ ✓ . 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - bash           ✓ ✓ ✓ . ✓ 

 Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

health#lspconfig#check
========================================================================
## Checking language server protocol configuration
  - ERROR: sumneko_lua: The given command "C:\\Users\\Andrew\\AppData\\Roaming\\nvim-data\\lsp_servers\\sumneko_lua\\extension\\server\\bin\\Windows\\lua-language-server" is not executable.
  - ERROR: tailwindcss: The given command "C:\\Users\\Andrew\\AppData\\Roaming\\nvim-data\\lsp_servers\\tailwindcss_npm\\node_modules\\.bin\\tailwindcss-language-server.cmd" is not executable.
  - INFO: null-ls: configuration checked.

health#nvim#check
========================================================================
## Configuration
  - WARNING: Missing user config file: C:\Users\Andrew\AppData\Local\nvim/init.vim
    - ADVICE:
      - :help |init.vim|

## Performance
  - OK: Build type: RelWithDebInfo

## Remote Plugins
  - OK: Up to date

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: win32yank

## Python 2 provider (optional)
  - WARNING: No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
  - ERROR: Python provider error:
    - ADVICE:
      - provider/pythonx: Could not load Python 2:
          python2 not found in search path or not executable.
          python2.7 not found in search path or not executable.
          python2.6 not found in search path or not executable.
          C:\Users\Andrew\AppData\Local\Programs\Python\Python39\python.EXE is Python 3.9 and cannot provide Python 2.
  - INFO: Executable: Not found

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3.9 in the environment.
  - INFO: Multiple python3.9 executables found.  Set `g:python3_host_prog` to avoid surprises.
  - INFO: Executable: C:\Users\Andrew\AppData\Local\Programs\Python\Python39\python3.9.EXE
  - INFO: Other python executable: C:/Users/Andrew/AppData/Local/Programs/Python/Python39/python3.9
  - INFO: Python version: 3.9.6
  - INFO: pynvim version: 0.4.3
  - OK: Latest pynvim is installed.

## Python virtualenv
  - OK: no $VIRTUAL_ENV

## Ruby provider (optional)
  - WARNING: `ruby` and `gem` must be in $PATH.
    - ADVICE:
      - Install Ruby and verify that `ruby` and `gem` commands work.

## Node.js provider (optional)
  - INFO: Node.js: v16.7.0
  - WARNING: Missing "neovim" npm (or yarn) package.
    - ADVICE:
      - Run in shell: npm install -g neovim
      - Run in shell (if you use yarn): yarn global add neovim

## Perl provider (optional)
  - WARNING: No usable perl executable found

health#treesitter#check
========================================================================
## Checking treesitter configuration
  - INFO: Runtime ABI version : 13
  - OK: Loaded parser for bash: ABI version 13
  - OK: Loaded parser for c: ABI version 13
  - OK: Loaded parser for css: ABI version 13
  - OK: Loaded parser for java: ABI version 13
  - OK: Loaded parser for javascript: ABI version 13
  - OK: Loaded parser for json: ABI version 13
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for python: ABI version 13
  - OK: Loaded parser for rust: ABI version 13
  - OK: Loaded parser for typescript: ABI version 13
  - OK: Loaded parser for yaml: ABI version 13
  - OK: Loaded parser for c: ABI version 13
  - OK: Loaded parser for bash: ABI version 13
  - OK: Loaded parser for c: ABI version 13
  - OK: Loaded parser for css: ABI version 13
  - OK: Loaded parser for java: ABI version 13
  - OK: Loaded parser for javascript: ABI version 13
  - OK: Loaded parser for json: ABI version 13
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for python: ABI version 13
  - OK: Loaded parser for rust: ABI version 13
  - OK: Loaded parser for typescript: ABI version 13
  - OK: Loaded parser for yaml: ABI version 13



[telescope] [WARN  17:14:20] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:20] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:20] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:21] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:21] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:21] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:22] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:22] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
[telescope] [WARN  17:14:22] ...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:442: Finder failed with msg:  ...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: rg: Executable not found
[telescope] stack traceback:
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:287: in function 'convert_opts'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:17: in function 'new'
[telescope] ^I...ite\pack\packer\start\telescope.nvim\lua\telescope\_.lua:59: in function 'spawn'
[telescope] ^I...elescope.nvim\lua\telescope\finders\async_job_finder.lua:48: in function 'finder'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:438: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437>
[telescope] ^I[C]: in function 'pcall'
[telescope] ^I...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:437: in function <...ck\packer\start\telescope.nvim\lua\telescope\pickers.lua:374>
```

### Screenshots

_No response_

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

### PR DESCRIPTION
**Pull request by kylo252**
# Description

- feat(windows): add custom config_win.example.lua
- ci(windows): first attempt at a workflow

Fixes #1672, #1804, #1823, #1902, #2309

### CODE REVIEW

**Comment by abzcoding** [comment]
@kylo252 wonderful work 👍 
can we address #2329 as well?

**Comment by kylo252** [comment]
> @kylo252 wonderful work 👍 can we address #2329 as well?

I was literally just trying to figure how to call that alias, it's hard to test in the CI 😢

**Comment by kylo252** [comment]
I have no idea how this makes sense....

image

**Comment by abzcoding** [comment]
are you planning to bring back the healthcheck or 😅

**Comment by kylo252** [comment]
> are you planning to bring back the healthcheck or 😅

This is still very hard unfortunately, it took me over two hours until I found out that `--headless` has some weird behavior on Windows. It's related to the result of this call where it includes both backslashes and forward slashes but only in `--headless` for some reason.https://github.com/tamago324/nlsp-settings.nvim/blob/7c73dfb04ab16c6ba495c3335d8afa24bb8c7e3b/lua/nlspsettings/schemas.lua#L9 

 I'm not even sure how to fix it reliably. Maybe we should skip the headless tests for now 😢

**Comment by abzcoding** [comment]
> > are you planning to bring back the healthcheck or 😅
> 
> This is still very hard unfortunately, it took me over two hours until I found out that `--headless` has some weird behavior on Windows. It's related to the result of this call where it includes both backslashes and forward slashes but only in `--headless` for some reason.https://github.com/tamago324/nlsp-settings.nvim/blob/7c73dfb04ab16c6ba495c3335d8afa24bb8c7e3b/lua/nlspsettings/schemas.lua#L9
> 
> I'm not even sure how to fix it reliably. Maybe we should skip the headless tests for now 😢

i mean we can skip headless tests for windows, if we are sure that the windows part works fine

**Comment by kylo252** [comment]
> i mean we can skip headless tests for windows, if we are sure that the windows part works fine

That's the problem, I'm not quite sure, and what makes it worse is that busted doesn't seem to work on windows either...

File: utils/installer/install.sh [hunk 1]
```diff
@@ -321,30 +321,29 @@ function verify_lvim_dirs() {
...
+    OS="$(uname -s)"
+    case "$OS" in
+      Linux | *BSD)
+        cp -r "$src/"* "$src.old/."
+        ;;
+      Darwin)
+        cp -R "$src/"* "$src.old/."
+        ;;
+      *)
+        echo "OS $OS is not currently supported."
```
Comment (abzcoding) [inline utils/installer/install.sh:343]: should we add an alternative for windows in here?

**Review by abzcoding — APPROVED** [review]
phew, thanks @kylo252 our windows support seems to be much better now 🎉 

LGTM 🚀

**Comment by kylo252** [comment]
@abzcoding, I'll add a note about busted not working in windows (at least not through plenary), otherwise, I need to figure out what's happening here: 
https://github.com/LunarVim/LunarVim/runs/5481539371?check_suite_focus=true#step:5:24

**Comment by kylo252** [comment]
@abzcoding, I think I've exhausted my resources here. I had just noticed that PackerSync wasn't being called originally and I couldn't get it to work either, but I assume it was always broken (same headless mode issue). 

Overall, hopefully this change will make things slightly easier to track issues, since the main goal was trying to cover all deps.

**Comment by abzcoding** [comment]
> @abzcoding, I think I've exhausted my resources here. I had just noticed that PackerSync wasn't being called originally and I couldn't get it to work either, but I assume it was always broken (same headless mode issue).
> 
> Overall, hopefully this change will make things slightly easier to track issues, since the main goal was trying to cover all deps.

Sounds good to me, dude. Thanks for all your efforts, merge it whenever you like 👍

**Pull request merged**